File: AUGUST-2025/01-08-2025/p118.swift
class Solution {
    func generate(_ numRows: Int) -> [[Int]] {
        guard numRows > 1 else { return [[1]] }

        var arr = [[1],[1,1]]

        for i in stride(from: 2, to: numRows, by: 1){
            arr.append([arr[i-1].first!] + Array(repeating: 0, count: arr[i-1].count - 1) + [arr[i-1].last!] ) 

            for j in 1..<arr[i].count - 1 {
                arr[i][j] = arr[i-1][j-1] + arr[i-1][j]
            }
        }

        return arr
    }
}

File: AUGUST-2025/01-08-2025/p118.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        if num_rows == 1 {vec![vec![1]]}
        else if num_rows == 2 {vec![vec![1], vec![1, 1]]}
        else {
            let mut v = vec![vec![1], vec![1, 1]];
            for  k in 3..=num_rows {
                let mut p = vec![1;k as usize];
                let t = v.last().unwrap();
                for i in 1..p.len()-1 {
                    p[i] = t[i-1] + t[i]
                }
                v.push(p);
            }
            v
        }
    }
}